### PR TITLE
N+1 solution and integration specs

### PR DESCRIPTION
**Pull request by utsargo**
### Description

- Fix N+1 problem
  - Before:
    Screenshot from 2023-04-18 16-18-10
  - After:
    Screenshot from 2023-04-18 23-39-45
- Add all integration specs

### Team
- @utsargo 
- @Dmambo

### CODE REVIEW

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑🛑
yPkfDFf

- [ ] Please consider removing the `Live demo` section from the readme file. It is only required when you have a project that is deployed. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/integration-specs/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      expect(page).to have_content(first_post.text[0..200])
+      expect(page).to have_content(second_post.text[0..200])
+    end
+    it 'should display recent comments on a post' do
+      expect(page).to have_content(first_comment.text)
+    end
+    it 'should display comment counts for each post' do
+      expect(page).to have_content("Comments: #{first_post.comments_counter}")
+      expect(page).to have_content("Comments: #{second_post.comments_counter}")
+    end
```
Comment (mikethreels) [inline spec/integration-specs/post_index_spec.rb:49]: - [x] according to the requirements for the Post index page the below tests should be included as well:
- Tests that asserts you can see how many likes a post has.
- Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
- Tests that redirects to the post's show page when you click on a post.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Utsargo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job on writing the tests

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @utsargo, and @Dmambo👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- All tests are passed. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**